### ISSUE TITLE
FieldArray not updates on own field change

### ISSUE THREAD

**Issue by vtaits** [erikras/redux-form#3008]
### Are you submitting a **bug report** or a **feature request**?
Bug

### What is the current behavior?
https://jsfiddle.net/vadim91/nduqzsst/1/

### What is the expected behavior?
In this example "Remove" button should disable other inputs and "Restore" button should enable them. But this doesn't happen after each button click. 

### What's your environment?
Ubuntu 16.04.2 LTS
node 6.10.1
npm 3.10.10

"react": "^15.5.4"
"react-dom": "^15.5.4"
"react-redux": "^5.0.5"
"redux": "^3.6.0"
"redux-form": "6.7.0"

### Other information
Looks like a bug in shouldComponentUpdate method of ConnectedFieldArray
```js
    shouldComponentUpdate(nextProps) {
      // Update if the elements of the value array was updated.
      const thisValue = this.props.value
      const nextValue = nextProps.value

      if (thisValue && nextValue) {
        if (
          thisValue.length !== nextValue.length ||
          thisValue.every(val => nextValue.some(next => deepEqual(val, next))) // <--------------
        ) {
          return true
        }
      }
```

Now component updates if each prev value equals each next value and not otherwise.

**Comment by erikras** [comment]
This is not a bug; it's a feature! 😀 The `FieldArray` component (`renderMembers` in this example) does not rerender every time one of the values inside it changes. That would be horribly inefficient for large arrays.

That said, the line that you highlighted makes no sense to me; I don't recall writing it, and all the tests pass when it is removed. Why on earth we would want to rerender the `FieldArray` when all the values are equal? It certainly seems misplaced.

As for your particular use case, I don't have a great answer. Perhaps some sort of prop like `rerenderOnEveryChange` to `FieldArray` might work....?

**Comment by erikras** [comment]
Fix published in [`v6.8.0`](https://github.com/erikras/redux-form/releases/tag/v6.8.0).

**Comment by vtaits** [comment]
Thank you!

**Comment by vtaits** [comment]
I caught a bug.
https://jsfiddle.net/nduqzsst/3/
If there are multiple fields groups with same values, "remove" works incorrectly.

A little later I'll fix.

**Comment by nihaux** [comment]
yep same problem happen when you change the order of the FieldArray.  
In my use case, I have an array of tags in which the order matters.   
If you switch tag 1 with tag 3 shouldComponentUpdateReturns false.  
Your PR https://github.com/erikras/redux-form/pull/3089 fixes it, thx 👍

**Comment by gustavohenke** [comment]
Since @nihaux said it fixes, I'll close this one. Watch out for new releases!

**Comment by vtaits** [comment]
Okay! Thank you!

**Comment by erikras** [comment]
Fix published in [`v7.0.0`](https://github.com/erikras/redux-form/releases/tag/v7.0.0).

**Comment by natalliakaranevich** [comment]
@erikras Thanks for update.
But it still not working for me if Field has type checkbox.
How can I fix it? Here is my code. I need uncheck all checkboxes in a **column** if new one in the column checked. But rerender not fires.
Please help. Thanks

`const renderDaily = ({ fields}) => {
    return <tbody>
    {
        fields.map((item, index) => {
                console.log('rendered');
                return <tr key={index}>
                    {
                        week.map((day, indexDay) => {
                            const name = `${item}.${day}`;

                            return <td key={indexDay}>
                                <input type="checkbox" defaultChecked={current[day]} name={name} />
                            </td>;
                        })
                    }
                </tr>;
            }
        )
    }
    </tbody>;
}`
